pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -551,9 +551,13 @@ func parseAppInstanceConfig(config *zconfig.EdgeDevConfig,
 		for _, adapter := range cfgApp.Adapters {
 			log.Tracef("Processing adapter type %d name %s",
 				adapter.Type, adapter.Name)
+			hwaddr, err := net.ParseMAC(adapter.EthVf.Mac)
+			if err != nil {
+				log.Errorf("Failed to parse hardware address for adapter: %s", adapter.Name)
+			}
 			appInstance.IoAdapterList = append(appInstance.IoAdapterList,
 				types.IoAdapter{Type: types.IoType(adapter.Type),
-					Name: adapter.Name})
+					Name: adapter.Name, EthVf: types.EthVF{Mac: hwaddr, VlanId: uint16(adapter.EthVf.VlanId), Logicallabel: adapter.EthVf.LogicalLabel}})
```

Comment:
It gets a bit hard to read this and it might make it more clear if there is a check on the type being VF before setting the VF-specific fields. E.g., 
 ioa := types.IoAdapter{Type: types.IoType(adapter.Type),
					Name: adapter.Name}
 if adapter.Type == VF { ...}
 appInstance.IoAdapterList = append(..., ioa)

Reply:
fixed